core/src/main/java/org/jdbi/v3/BuiltInArgumentFactory.java

```diff
@@ -89,10 +94,25 @@ public Argument build(Class<?> expectedType, T value, StatementContext ctx)
         }
 
         // Enums must be bound as VARCHAR.
-        if (expectedType.isEnum()) {
+        if (expectedClass.isEnum()) {
             return new StringArgument(value.toString());
         }
 
+        if (value instanceof Optional) {
```

Comment:
Maybe?  Not so convinced now that I think about it more.  I just hate the "chained special conditions" feeling of doing it first for Enum, then Optional, then ...